### PR TITLE
network added

### PR DESCRIPTION
**Pull request by Mayar-Abdulkareem**

### CODE REVIEW

File: GP_iOS/Constants/NetworkConstant.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+//  GP_iOS
+//
+//  Created by FTS on 06/11/2023.
+//
+
+import Foundation
+
+struct NetworkConstant {
+    static let loginBaseURL = ""
+    static var baseURL = ""
```
Comment (BassamHillo1997) [inline GP_iOS/Constants/NetworkConstant.swift:12]: The server should have one base URL per instance and each API should have it's own path

File: GP_iOS/Network/Clients/BaseClient.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    
+    /**
+     Performs a network request using Alamofire.
+     - Parameters:
+     - router: The router that defines the API endpoint, method, and parameters.
+     - completion: A closure to be executed when the request completes, providing a `Result` with the decoded data or an `AFError` in case of failure.
+     */
+    func performRequest<T: Codable> (
+        router: BaseRouter,
+        completion: @escaping (Result<T, AFError>) -> ()) {
```
Comment (BassamHillo1997) [inline GP_iOS/Network/Clients/BaseClient.swift]: fix indent

File: GP_iOS/Network/Routers/Base/BaseUnitRouter.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+//  BaseUnitRouter.swift
+//  GP_iOS
+//
+//  Created by FTS on 13/11/2023.
+//
+
+import Foundation
+import Alamofire
+
+protocol BaseUnitRouter: BaseRouter {}
```
Comment (BassamHillo1997) [inline GP_iOS/Network/Routers/Base/BaseUnitRouter.swift]: why you created this empty protocol ?

File: GP_iOS/Network/Routers/Layers/LoginLayer.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
+//
+//  UserRouter.swift
```
Comment (NouraHashem-fts) [inline GP_iOS/Network/Routers/Layers/LoginLayer.swift]: Rename to LoginRouter and fileName as well

**Pull request merged**